src/components/TradeWidget/TokenRow.tsx

```diff
@@ -140,15 +197,29 @@ const TokenRow: React.FC<Props> = ({ token, tokens, selectLabel, onSelectChange
         />
       </SelectBox>
       <InputBox>
-        <input type="text" placeholder="0" />
+        <input
+          className={className}
+          placeholder="0"
+          name={inputId}
+          type="text"
+          required
+          ref={register({
+            validate: {
+              positive: (value: string): true | string => validatePositive(value),
+            },
+          })}
+        />
+        {errorOrWarning}
         <WalletDetail>
           <strong>
             <Link to="/deposit">Exchange wallet:</Link>
           </strong>{' '}
-          <span className="success">312312.33</span>
+          <a className="success" onClick={(): void => setValue(inputId, max.toFixed(4))}>
+            {formatAmount((balance || {}).exchangeBalance, token.decimals) || 0}
```

Comment:
Should clicking the exchange wallet amount really fill max? seems a bit dangerous imo

Reply:
Not sure. In my view it's convenient but no strong opinion here. I'm fine not having it either.
This is not part of the requirements, but since we do have a way to fill in the max when depositing or withdrawing, I thought would be useful to have it here as well.

@anxolin what do you say?